test/test_opaque_obj_v2.py

```diff
@@ -70,10 +72,43 @@ def forward(self, x, y):
         return x * y
 
 
+class ValueConfig:
+    def __init__(self, mode: str):
+        self.mode = mode
+
+    def __eq__(self, other):
+        return isinstance(other, ValueConfig) and self.mode == other.mode
+
+    def __hash__(self):
+        return hash(self.mode)
+
+    def __repr__(self):
+        return f"ValueConfig(mode={self.mode!r})"
+
+
+class SizeStore:
+    def __init__(self, size: str):
+        self.size = size
+
+    def __eq__(self, other):
+        return isinstance(other, ValueConfig) and self.size == other.size
+
+    def __hash__(self):
+        return hash(self.size)
+
+    def __repr__(self):
+        return f"SizeStore(size={self.size!r})"
+
+    def increment_size(self):
+        return self.size + 1
+
+
 register_opaque_type(OpaqueQueue)
 register_opaque_type(RNGState)
 register_opaque_type(Counter)
 register_opaque_type(AddModule)
+register_opaque_type(ValueConfig, value_type=True)
```

Comment:
nit: I prefer prefer typ="value"/"reference" over `value_type=True`, this implies that a non-value type is always a reference type (which I'm not sure is true in general?)